File: problems/problems_3201/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.maximumLength(test_input)

    def maximumLength(self, nums: List[int]) -> int:
        dp = [0] * 4
        for num in nums:
            cur = num % 2
            for i in range(4):
                if (i >> (dp[i] % 2)) & 1 == cur:
                    dp[i] += 1
        return max(dp)

File: problems/problems_3201/Solution.java
package problems.problems_3201;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int maximumLength(int[] nums) {
        int ans = 0;
        int[] dp = new int[4];
        for (int num: nums) {
            int cur = num & 1;
            for (int i = 0; i < 4; ++i) {
                if (((i >> (dp[i] & 1)) & 1) == cur) {
                    ans = Math.max(ans, ++dp[i]);
                }
            }
        }
        return ans;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] nums = jsonArrayToIntArray(inputJsonValues[0]);
        return JSON.toJSON(maximumLength(nums));
    }
}
